airbyte-cdk/python/airbyte_cdk/sources/message/repository.py

```diff
@@ -62,10 +70,47 @@ def emit_message(self, message: AirbyteMessage) -> None:
             raise ValueError(f"As of today, only {_SUPPORTED_MESSAGE_TYPES} are supported as part of the InMemoryMessageRepository")
         self._message_queue.append(message)
 
-    def log_message(self, level: Level, message_provider: Callable[[], AirbyteMessage]) -> None:
-        if self._SEVERITY_BY_LOG_LEVEL[self._log_level] >= self._SEVERITY_BY_LOG_LEVEL[level]:
-            self.emit_message(AirbyteMessage(type=Type.LOG, log=AirbyteLogMessage(level=level, message=message_provider())))
+    def log_message(self, level: Level, message_provider: Callable[[], LogMessage]) -> None:
+        if _is_severe_enough(self._log_level, level):
+            self.emit_message(
+                AirbyteMessage(type=Type.LOG, log=AirbyteLogMessage(level=level, message=filter_secrets(json.dumps(message_provider()))))
+            )
 
     def consume_queue(self) -> Iterable[AirbyteMessage]:
         while self._message_queue:
             yield self._message_queue.pop(0)
+
+
+class LogAppenderMessageRepositoryDecorator(MessageRepository):
+    def __init__(self, dict_to_append: LogMessage, decorated: MessageRepository, log_level=Level.INFO):
+        self._dict_to_append = dict_to_append
+        self._decorated = decorated
+        self._log_level = log_level
+
+    def emit_message(self, message: AirbyteMessage) -> None:
+        self._decorated.emit_message(message)
+
+    def log_message(self, level: Level, message_provider: Callable[[], LogMessage]) -> None:
+        if _is_severe_enough(self._log_level, level):
+            message = message_provider()
+            self._append_second_to_first(message, self._dict_to_append)
+            self._decorated.log_message(level, lambda: message)
+
+    def consume_queue(self) -> Iterable[AirbyteMessage]:
+        return self._decorated.consume_queue()
+
+    def _append_second_to_first(self, first, second, path=None):
+        if path is None:
```

Comment:
It is recommended that default values be immutable since multiple method calls use the same reference. For example:

```
def x(arg=[]):
    arg.append(1)
    print(arg)
    
x()
x()
```

...will output:
```
[1]
[1, 1]
```

Therefore, I would prefer to ensure that we use an immutable value to avoid such confusion